Automattic/jetpack
# Protect: update the Status when connection is established

#### Changes proposed in this Pull Request:

* Updates the Protect status after the connection is established
* Sort the themes and plugins list leaving items with vulnerabilities first and not checked items second

#### Jetpack product discussion


NA

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Start with a disconnected site
* Go to Jetpack Protect and open the cosole
* Click "Get started with Protect"
* Confirm there's a request made to `jetpack-protect/v1/status` after the connection is established
* Confirm the list is updated with the correct vulnerabilities and all information

Check order:
* Set response_type to `incomplete`
* Make sure the plugin marked as `notChecked` are sorted in the list right below the items with vulnerabilities - and above items with no vulnerabiltiies
* Set reponse_type to `incomplete_green`
* Make the plugin marked as `notChecked` are the first in the list

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - 0-as-1202161351799999
